events/src/gateway.rs

```diff
@@ -0,0 +1,9 @@
+use serde::{Deserialize, Serialize};
+
+pub type Version = String;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum Event {
+    Online(Version),
+    Offline,
+}
```

Comment:
The default serialization for this is going to produce `{ "Online": "<version>" }` for online and `{ "Offline": null }` AFAICT.

I would much prefer internal tagging plus the version being a struct variant, so get `{ "status": "online", "version": "<version>" }` and `{ "status": "offline" }`:

```suggestion
#[serde(tag = "status", rename_all = "lowercase")]
pub enum Event {
    Online { version: String },
    Offline,
}
```